### ISSUE TITLE
Washed out colors Mac OS

### ISSUE THREAD

**Issue by ben-cl** [moonlight-stream/moonlight-qt#892]
**Describe the bug**
I get washed out colors on my MacBook Pro 14 2021. 
I always had this issue using hardware decoding or H.265. I did found a reddit post in the past saying to use Software Decoding + H264 and it was working for me (the only option fixing the color space). 

Today even this option is not working anymore. This is why I am creating the post. I am unsure if it is since the last macOS update Ventura 13.0. 

I did try downloading older version of moonlight but I do not seems to get success. 

**Steps to reproduce**
Any special steps that are required for the bug to appear.

Force software decoding and H.264 to see if color space is working. 

**Screenshots**
If applicable, add screenshots to help explain your problem. If the issue is related to video glitching or poor quality, please include screenshots.

I had this screenshot of my game in the past when the color space was working :

<img width="1512" alt="Screen Shot 2022-07-27 at 5 28 08 PM" src="https://user-images.githubusercontent.com/32286421/199872569-4e584450-2ca2-41c6-9333-923ba56c1ada.png">
<img width="1512" alt="Screen Shot 2022-07-27 at 5 24 36 PM" src="https://user-images.githubusercontent.com/32286421/199874146-e55762f9-562e-4901-9733-b5ffffd4943c.png">

This is a screenshot of my game today at the same spot with washed out color. I am using the same setting: 

<img width="1512" alt="Screenshot 2022-11-03 at 10 17 28 PM" src="https://user-images.githubusercontent.com/32286421/199872731-d6bb0222-3217-440b-8fb1-f04d4d6e4aef.png">

**Affected games**
List the games you've tried that exhibit the issue. To see if the issue is game-specific, try streaming Steam Big Picture with Moonlight and see if the issue persists there.

Windows itself, emulators, games. 

**Client PC details (please complete the following information)**
 - OS: Mac OS 13.0 Ventura
 - Moonlight Version: 4.3.1, 4.3.0 and older versions

**Server PC details (please complete the following information)**
 - OS: Windows 11 21H2
 - GeForce Experience version: 3.26.0.154
 - Nvidia GPU driver: 526.47
 - Antivirus and firewall software: Kaspersky 

**Moonlight Logs (please attach)**

[Moonlight-1667527291.log](https://github.com/moonlight-stream/moonlight-qt/files/9934579/Moonlight-1667527291.log)

**Additional context**
Anything else you think may be relevant to the issue

**Comment by vhd** [comment]
I also noticed this a long time ago. But after a lot of troubleshooting, I came to the conclusion that actually **it is oversaturated in software mode**
When I put the same image on the same 4K monitor connected to host it looks almost identical to hardware h265 decoded image but oversaturated with software decoding.
Also, there are some applications (usually it's SDL2 apps) that look oversaturated when they run natively on mac. For example Steam, DosBox and... Moonlight software decoder (because it uses SDL)
You can see some examples here
https://github.com/moonlight-stream/moonlight-qt/issues/493#issuecomment-1005825740
Or try to experiment yourself by taking screenshots with snippet tool on host and then opening it on Mac

**Comment by Glaceon2** [comment]
>  it is oversaturated in software mode

@vhd Why do images on Mac appear washed out if moonlight (or Mac) oversaturates them? I'm guessing the color would be more vibrant. Also, I thought the color-mismatch issue had been resolved because there had been several patches to moonlight, but this post makes me doubt it. Can you provide more information on the current state of the problem?

**Comment by ben-cl** [comment]
@vhd @Glaceon2 

I think you might be right @vhd ! I did test again today on Ventura 13.1 and software decoding seems to apply the saturation like before (which I prefer). 

Video on my mac:
![mac](https://user-images.githubusercontent.com/32286421/208806820-e0c54b82-3142-4a07-bb97-92ca649453e3.jpg)

Hardware H.265 - Same video and color seems to match:
![V1](https://user-images.githubusercontent.com/32286421/208807110-f63fb121-a1b3-44ed-9bc5-5977b1d6fa88.jpg)

Software H.264 - More saturation:
![2](https://user-images.githubusercontent.com/32286421/208807314-ddc71b3f-5bd7-4c53-b418-63e8e9c09fe7.jpg)

@vhd Do you think it is possible to apply SDL saturation with hardware decoding? Like even the chrome logo on my taskbar look much better in H.264 in my opinion.  

Thanks for the reply indeed I believe you are right.

**Comment by vhd** [comment]
@ben-cl You can try to play around with different color profiles under display settings
On my MBP 14" M1 pro there is "HDTV video" option, which looks even better (IMO)
On my older MacBook there were color profiles from SAMSUNG TV and LG monitor (I don't remember which one I actually used before)

**Comment by ben-cl** [comment]
I see, thanks for the tips! Will closed the thread since the color are actually correct in hardware decoding.

**Comment by achaiah** [comment]
I've been having the same issue on an M1 MacBook (with an external monitor) for the past 6 mo. or so. From this thread I don't necessarily get what the solution is. The colors are always washed out no matter which encoding option I select. They're definitely NOT washed out on the actual server I'm streaming from. Any ideas?

**Comment by paya-cz** [comment]
> I've been having the same issue on an M1 MacBook (with an external monitor) for the past 6 mo. or so. From this thread I don't necessarily get what the solution is. The colors are always washed out no matter which encoding option I select. They're definitely NOT washed out on the actual server I'm streaming from. Any ideas?

Just came to this thread after having issues on my MacBook with 2 external monitors. This is likely not a problem with Moonlight but with your setup. I managed to fix this via:

- On MacBook screen, bump up the brightness. You might have `Automatically adjust brightness` setting turned on, which will increase/decrease brightness for you based on ambient light. Play with these settings and see what makes the difference for you.
- On a portable USB-C external monitor, I actually switched a colour profile (MacOS -> Settings app -> Displays). `Adobe RGB (1998)` worked great for the monitor I have. Then I bumped the brightness on the monitor's OSD.
- For a second external monitor, I also switched a colour profile in MacOS settings, this time to `sRGB IEC61966-2.1`. At the same time, I used monitor's OSD to bump up the brightness as well as change a color profile within the monitor's OSD. Color temperature `Normal` worked good for me, but this will differ based on each make and model you might have. You will need to experiment here. Use your MacBook screen as a reference point, calibrate over multiple different images.
